src/validationinterface.cpp

```diff
@@ -187,7 +187,7 @@ void ThreadNotifyWallets(CBlockIndex *pindexLastTip)
             // Let wallets know transactions went from 1-confirmed to
             // 0-confirmed or conflicted:
             for (const CTransaction &tx : block.vtx) {
-                SyncWithWallets(tx, NULL);
+                SyncWithWallets(tx, NULL, chainActive.Height());
```

Comment:
We need to call `chainActive.Height()` within the above `LOCK(cs_main)` context and store the height in a local, rather than calling it here.